pkg/controller/logstash/config_test.go

```diff
@@ -110,6 +139,29 @@ log:
 			},
 			wantErr: true,
 		},
+		{
+			name: "logstash config disables TLS and service enables TLS",
+			args: args{
+				runtimeObjs: nil,
+				logstash: v1alpha1.Logstash{
+					Spec: v1alpha1.LogstashSpec{
+						Config: &commonv1.Config{Data: map[string]interface{}{
+							"api.ssl.enabled": "false",
+						}},
+						Services: []v1alpha1.LogstashService{{
+							Name: LogstashAPIServiceName,
+							TLS: commonv1.TLSOptions{
+								SelfSignedCertificate: &commonv1.SelfSignedCertificate{
+									Disabled: false,
```

Comment:
This test passes regardless of whether the value here has the TLS block, or whether `selfSignedCertificate.Disabled` value is `true` or `false`, as only the `useTLS` flag is relevant here. 

I think we are missing a layer of testing, of how the `useTLS` setting is derived from the `TLSOptions`